forge/csrc/passes/commute_utils.cpp

```diff
@@ -1008,6 +1008,19 @@ void restore_bcast_on_condition(
     }
 }
 
+int update_bcast_dim_commuted_through_transpose(int dim, graphlib::OpNode *op)
```

Comment:
Can we add a unit test to cover added function/logic? 

E.g. similar to other tests for this pass `forge/csrc/test/passes/test_erase_inverse_ops.cpp`

Reply:
Done. Added the same case we reproed in #546